### ISSUE TITLE
issue: Problem with regex after upgrade to v1.82.0

### ISSUE THREAD

**Issue by Vladikamira** [VictoriaMetrics/VictoriaMetrics#3227]
**Describe the bug**
after an upgrade from `v1.78.1` to `v1.82.0` we faced weird bug, when some regular expressions didn't work properly. For instance 
regex `status=~"(499|5..)"` could give results with `status="200"`.
Rollback to `v1.78.1` helped with such an issue. We see no such problem on LTS version `v1.79.4`, so we stayed there.
During tests upgrades later, we could see that problem solves in 1-2 hours, fyi

**To Reproduce**
Upgrade to `v1.82.0` and check regular expressions

**Expected behavior**
regex works properly

**Screenshots**
<img width="1421" alt="Screenshot 2022-10-10 at 16 53 58" src="https://user-images.githubusercontent.com/5423271/195304752-5c52059c-ce97-4b02-a100-df274e67bd2a.png">
![image](https://user-images.githubusercontent.com/5423271/195304763-d17b166a-421a-4f6e-9ab4-8d8b2317e2a3.png)

**Version**
```
/ # /vmstorage-prod --version
vmstorage-20221007-005746-tags-v1.82.0-cluster-0-g58ee1b9ed
```

**Used command-line flags**
vmstorage
```
    Args:
      --retentionPeriod=6
      --storageDataPath=/storage
      --dedup.minScrapeInterval=15s
      --envflag.enable=true
      --envflag.prefix=VM_
      --http.maxGracefulShutdownDuration=10s
      --http.shutdownDelay=5s
      --loggerFormat=json
      --loggerLevel=INFO
      --loggerOutput=stdout
      --memory.allowedPercent=60
      --search.maxUniqueTimeseries=30000000
```

vminsert
```
    Args:
      --storageNode="ADDRESS_1"
      --storageNode="ADDRESS_2"
      --clusternativeListenAddr=:8400
      --disableRerouting=true
      --envflag.enable=true
      --envflag.prefix=VM_
      --import.maxLineLen=104857600
      --insert.maxQueueDuration=1m
      --loggerFormat=json
      --loggerLevel=INFO
      --loggerOutput=stdout
      --maxConcurrentInserts=8
      --maxInsertRequestSize=33554432
      --maxLabelsPerTimeseries=50
      --memory.allowedPercent=60
      --replicationFactor=1
      --vmstorageDialTimeout=1s
```
vmselect
```
    Args:
      --cacheDataPath=/cache
      --storageNode="ADDRESS_1"
      --storageNode="ADDRESS_2"
      --dedup.minScrapeInterval=15s
      --envflag.enable=true
      --envflag.prefix=VM_
      --http.disableResponseCompression=false
      --http.maxGracefulShutdownDuration=10s
      --http.shutdownDelay=5s
      --loggerFormat=json
      --loggerLevel=INFO
      --loggerOutput=stdout
      --memory.allowedPercent=60
      --replicationFactor=1
      --search.denyPartialResponse=true
      --search.logSlowQueryDuration=5s
      --search.maxConcurrentRequests=500
      --search.maxExportSeries=100000000
      --search.maxQueryDuration=115s
      --search.maxQueryLen=131072
      --search.maxQueueDuration=100ms
      --search.maxSamplesPerQuery=10000000000
      --search.maxSeries=1000000
      --search.maxStalenessInterval=30s
      --search.maxUniqueTimeseries=100000000
      --search.queryStats.minQueryDuration=1s
      --vmstorageDialTimeout=1s
```

**Comment by f41gh7** [comment]
Hello, thanks for reporting. 

Cannot reproduce it for now at least for single node version. Probably it's cluster specific or cache corruption. Need to check it additionally with cluster.

**Comment by valyala** [comment]
Confirmed the bug - it should be fixed in the commit ed324aad66177e523611873a10c5b9ad961a2863 . This commit will be included in the next release. In the mean time it is possible to check whether the commit fixes the issue by building `vmstorage` from the commit ed324aad66177e523611873a10c5b9ad961a2863 according to [these docs](https://docs.victoriametrics.com/Cluster-VictoriaMetrics.html#building-from-sources) and verifying whether the issue stops reproducing there.

**Comment by valyala** [comment]
The bug has been fixed in [VictoriaMetrics v1.82.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Note that this bug also may concern [relabeling](https://docs.victoriametrics.com/vmagent.html#relabeling) at vmagent.
